sdk/src/main/java/io/radar/sdk/RadarVerificationManager.kt

```diff
@@ -18,10 +29,322 @@ internal class RadarVerificationManager(
+                                                        beacons: Array<RadarBeacon>?
+                                                    ) {
+                                                        if (status != Radar.RadarStatus.SUCCESS || beacons == null) {
+                                                            callTrackApi(null)
+
+                                                            return
+                                                        }
+
+                                                        callTrackApi(beacons)
+                                                    }
+                                                })
+                                        } else if (beacons != null) {
+                                            Radar.beaconManager.startMonitoringBeacons(beacons)
+
+                                            Radar.beaconManager.rangeBeacons(
+                                                beacons,
+                                                false,
+                                                object : Radar.RadarBeaconCallback {
+                                                    override fun onComplete(
+                                                        status: Radar.RadarStatus,
+                                                        beacons: Array<RadarBeacon>?
+                                                    ) {
+                                                        if (status != Radar.RadarStatus.SUCCESS || beacons == null) {
+                                                            callTrackApi(null)
+
+                                                            return
+                                                        }
+
+                                                        callTrackApi(beacons)
+                                                    }
+                                                })
+                                        } else {
+                                            callTrackApi(null)
+                                        }
+                                    }
+                                },
+                                false
+                            )
+                        } else {
+                            callTrackApi(null)
+                        }
+                    }
+                }
+            })
+    }
+
+    fun startTrackingVerified(token: Boolean, interval: Int, beacons: Boolean) {
+        val verificationManager = this
+
+        val request = NetworkRequest.Builder()
+            .addTransportType(NetworkCapabilities.TRANSPORT_ETHERNET)
+            .addTransportType(NetworkCapabilities.TRANSPORT_CELLULAR)
+            .addTransportType(NetworkCapabilities.TRANSPORT_BLUETOOTH)
+            .addTransportType(NetworkCapabilities.TRANSPORT_WIFI)
+            .removeCapability(NetworkCapabilities.NET_CAPABILITY_NOT_VPN)
+            .build()
```

Comment:
this is really cool, but I'll want to noodle on this more, especially cause we're not doing anything analogous on iOS

Reply:
We are! Here's the iOS equivalent: https://github.com/radarlabs/radar-sdk-ios/pull/279/files#diff-b06deb2cd9256c7428f1b0ff9382c61ae289d7deb9238ea1cfec16ecf2d6deebR252

Further reading: https://msolarana.netlify.app/2018/08/04/monitoring-network-changes/

Reply:
Ahh yes, just saw, nice.